File: Day1/2monotonic_array.js
function monotonic_array(a) {
  array_length = a.length;
  first_number = a[0];
  last_number = a[array_length - 1];

  if (first_number < last_number) {
    for (let i = 0; i < array_length - 1; i++) {
      if (a[i] > a[i + 1]) {
        return false;
      }
    }
  } else if (first_number == last_number) {
    for (let i = 0; i < array_length - 1; i++) {
      if (a[i] != a[i + 1]) {
        return false;
      }
    }
  } else {
    for (let i = 0; i < array_length - 1; i++) {
      if (a[i] < a[i + 1]) {
        return false;
      }
    }
  }
  return true;
}

console.log(monotonic_array([3, 4, 6]));
console.log(monotonic_array([3, 3, 3]));
console.log(monotonic_array([3, 1, -3]));
console.log(monotonic_array([3, 1, -3, 2]));

File: Day1/2monotonic_array.py
def monotonic_Array(a):
    n = len(a)
    first_number = a[0]
    last_number = a[n-1]

    if first_number<last_number:
        for i in range(n-1):
            if a[i] > a[i+1]:
                return False
    elif first_number == last_number:
        for i in range(n-1):
            if a[i] != a[i+1]:
                return False
    else:
        for i in range(n-1):
            if a[i] < a[i+1]:
                return False
    return True

print(monotonic_Array([3, 4, 5, 6, 7]))
print(monotonic_Array([3, 3, 3, 3, 3]))
print(monotonic_Array([3, 2, 1, -2, -3]))
print(monotonic_Array([3, 2, 1, -2, 4]))